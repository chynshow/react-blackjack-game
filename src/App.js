import React, { useContext, useEffect } from 'react';
import GameTitle from './components/GameTitle';
import MainActionPanel from './components/MainActionPanel';
import InfoPanel from './components/InfoPanel';
import GameTable from './components/GameTable';
import ActionPanel from './components/ActionPanel';
import { AppContext } from './state/AppContext';
import BetForm from './components/BetForm';
import Loader from './components/Loader';
import InfoModal from './components/InfoModal/index';
import RoundHistory from './components/InfoPanel/RoundHistory';

const App = () => {
  const {
    initApp,
    // saveGame,
    state: { bet, roundStarted, gameStarted, loading, roundHistory },
  } = useContext(AppContext);
  useEffect(() => {
    initApp();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  // const handleOnClose = (e) => {
  //   e.preventDefault();
  //   e.returnValue = '';
  //   saveGame();
  // };

  // useEffect(() => {
  //   window.addEventListener('beforeunload', (e) => handleOnClose(e));
  //   return () => {
  //     window.removeEventListener('beforeunload', (e) => handleOnClose(e));
  //   };
  // }, []);

  return (
    <>
      {loading ? (
        <Loader />
      ) : (
        <main className='c-main-container'>
          <GameTitle />
          <MainActionPanel />
          {gameStarted && <InfoPanel />}

          <div className='c-main-container__aligner'>
            <RoundHistory roundHistory={roundHistory} />
            <GameTable />
          </div>
          {gameStarted && !roundStarted && bet <= 0 && <BetForm />}
          {bet > 0 && <ActionPanel />}
          <InfoModal />
        </main>
      )}
    </>
  );
};

export default App;
