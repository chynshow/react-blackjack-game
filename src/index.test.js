import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

it('render App component', () => {
  const c = shallow(<App />);
  expect(c).toMatchSnapshot();
});
