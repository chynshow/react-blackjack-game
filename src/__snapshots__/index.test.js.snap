// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App component 1`] = `
<Fragment>
  <main
    className="c-main-container"
  >
    <GameTitle />
    <MainActionPanel />
    <div
      className="c-main-container__aligner"
    >
      <Memo(RoundHistory)
        roundHistory={Array []}
      />
      <GameTable />
    </div>
    <InfoModal />
  </main>
</Fragment>
`;
